### PR TITLE
EES-1258 Ensure ampersands encoded correctly when passed as url params.

### PR DESCRIPTION
**Pull request by shakes90**

### CODE REVIEW

File: src/explore-education-statistics-admin/src/services/releaseDataFileService.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ const releaseDataFileService = {
     data.append('metaFile', request.metadataFile);
 
     const file = await client.post<DataFileInfo>(
-      `/release/${releaseId}/data?name=${request.name}`,
+      `/release/${releaseId}/data?name=${encodeURIComponent(request.name)}`,
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/services/releaseDataFileService.ts:104]: I don't think we need to actually do this directly, You can pass a third `options` param to `client.post`, which can contain some `params` (query parameters). Really we should be doing this in lots of other places as well.

File: src/explore-education-statistics-admin/src/services/releaseAncillaryFileService.ts [hunk 2]
```diff
@@ -51,7 +51,9 @@ const releaseAncillaryFileService = {
     data.append('file', request.file);
 
     const file = await client.post<AncillaryFileInfo>(
-      `/release/${releaseId}/ancillary?name=${request.name}`,
+      `/release/${releaseId}/ancillary?name=${encodeURIComponent(
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/services/releaseAncillaryFileService.ts:54]: I don't think we need to actually do this directly, You can pass a third `options` param to `client.post`, which can contain some `params` (query parameters). Really we should be doing this in lots of other places as well.

File: src/explore-education-statistics-admin/src/services/releaseDataFileService.ts [hunk 3]
```diff
@@ -115,7 +115,7 @@ const releaseDataFileService = {
     data.append('zipFile', request.zipFile);
 
     const file = await client.post<DataFileInfo>(
-      `/release/${releaseId}/zip-data?name=${request.name}`,
+      `/release/${releaseId}/zip-data?name=${encodeURIComponent(request.name)}`,
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/services/releaseDataFileService.ts:118]: I don't think we need to actually do this directly, You can pass a third `options` param to `client.post`, which can contain some `params` (query parameters). Really we should be doing this in lots of other places as well.

File: src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseFileUploadsSection.tsx [hunk 4]
```diff
@@ -36,6 +36,8 @@ const errorMappings = [
       FILE_TYPE_INVALID: 'Choose a file of an allowed format',
       FILENAME_CANNOT_CONTAIN_SPACES_OR_SPECIAL_CHARACTERS:
         'Filename cannot contain spaces or special characters',
+      FILE_UPLOAD_NAME_CANNOT_CONTAIN_SPECIAL_CHARACTERS:
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseFileUploadsSection.tsx:39]: This is targetting the wrong form field. It should be mapped against `name`.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileUploadsValidatorService.cs [hunk 5]
```diff
@@ -48,7 +48,8 @@ public FileUploadsValidatorService(ISubjectService subjectService, IFileTypeServ
                 .OnSuccess(async _ => await ValidateSubjectName(releaseId, name));
         }
 
-        public async Task<Either<ActionResult, Unit>> ValidateFileForUpload(Guid releaseId, IFormFile file, ReleaseFileTypes type, bool overwrite)
+        public async Task<Either<ActionResult, Unit>> ValidateFileForUpload(Guid releaseId, IFormFile file,
+            ReleaseFileTypes type, bool overwrite, string? name = "")
```
Comment (ntsim) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileUploadsValidatorService.cs:52]: Name can be nullable?

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-admin/src/services/releaseDataFileService.ts (hunk 1 above)
Comment (shakes90) [reply src/explore-education-statistics-admin/src/services/releaseDataFileService.ts]: done - cheers

File: src/explore-education-statistics-admin/src/services/releaseAncillaryFileService.ts (hunk 2 above)
Comment (shakes90) [reply src/explore-education-statistics-admin/src/services/releaseAncillaryFileService.ts]: done

File: src/explore-education-statistics-admin/src/services/releaseDataFileService.ts (hunk 3 above)
Comment (shakes90) [reply src/explore-education-statistics-admin/src/services/releaseDataFileService.ts]: done

File: src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseFileUploadsSection.tsx (hunk 4 above)
Comment (shakes90) [reply src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseFileUploadsSection.tsx]: Yep - well spotted :) - done

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileUploadsValidatorService.cs (hunk 5 above)
Comment (shakes90) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileUploadsValidatorService.cs]: OK - have added a separate method to validate the file upload name as probably clearer that way.

**Review by ntsim — APPROVED** [review]

**Pull request merged**